straxen/plugins/peaklet_processing.py

```diff
@@ -601,47 +604,80 @@ def compute(self, peaklets):
     @numba.njit(cache=True, nogil=True)
     def get_merge_instructions(
             peaklet_starts, peaklet_ends, areas, types,
-            cluster_starts, cluster_stops,
-            max_duration, max_area):
-        start_merge_at = np.zeros(len(cluster_starts), dtype=np.int32)
-        end_merge_at = np.zeros(len(cluster_starts), dtype=np.int32)
-        n_to_merge = 0
-        left_i = 0
-
-        for cluster_i, cluster_start in enumerate(cluster_starts):
-            cluster_stop = cluster_stops[cluster_i]
+            gap_thresholds, max_duration, max_gap, max_area):
+        """Finding the group of peaklets to merge
+        returns: list of the first index of peaklet to be merged and
+        list of the exclusive last index of peaklet to be merged
+        """
+        peaklet_gaps = peaklet_starts[1:] - peaklet_ends[:-1]
+        n_gaps = len(peaklet_gaps)
+
+        previous_valid_gap = np.arange(n_gaps) - 1
```

Comment:
Thanks Tianyu, I really had to make a sketch to keep track of all the shifts by +1 or -1 and go over it. I think it looks good but it's a bit hard to fully keep track of all those indices.

Reply:
Yes, I have a sketch that I should have put in the pr first. But not sure how much commenting should be in the code, or just explain this in the pr is enough.